tests/benchmark/crud_test.go

```diff
@@ -0,0 +1,895 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func getRootResource() string {
+	registryProject, valueExists := os.LookupEnv("REGISTRY_PROJECT_IDENTIFIER")
+	if valueExists {
+		return fmt.Sprintf("projects/%s/locations/global", registryProject)
+	}
+	return "projects/bench/locations/global"
+}
+
+func getNumberOfVersions() int {
+	strNumberOfVersions, exists := os.LookupEnv("REGISTRY_VERSION_COUNT")
+	numberOfVersions, err := strconv.Atoi(strNumberOfVersions)
+	if err != nil || !exists {
+		numberOfVersions = 1
+	}
+	return numberOfVersions
+}
```

Comment:
Nit: Function names starting with "get" aren't very common in Go. Usually functions like `getFoo` are shortened to just `foo`.

In cases like this where the value isn't readily available, the function name can usually include a different verb to indicate it's coming from somewhere else, e.g. `fetchFoo`, `generateBar`, etc. For this particular case I recommend something like `loadVersionCount`.

**More importantly**: I think we should fail the test if the input count is invalid to communicate the problem to the user. We want them to know we're not proceeding with whatever they tried to set the value to.

```
func loadVersionCount(t *testing.T) {
	t.Helper()

	s, ok := os.LookupEnv("REGISTRY_VERSION_COUNT")
	if !ok {
		return 1 // Default to a single version.
	}

	c, err := strconv.Atoi(s)
	if err != nil {
		t.Fatalf("Setup: Invalid REGISTRY_VERSION_COUNT %q: count must be an integer", s)
	}

	return c
}
```